### ISSUE TITLE
Network*online* not respected by systemd service on boot

### ISSUE THREAD

**Issue by jobs-git** [systemd/systemd#19786]
In NFS for example, it always fails on boot due to `network not reachable`. The only solution that worked was replace the `network*online*` with a custom script that checks the network.

I am led to believe that when the `network*online*` unit says its `online` when in fact its just `connected` thus there is no guarantee that a network communication between parties can be established. 

See https://unix.stackexchange.com/questions/554961/nfs-mount-fails-on-boot-although-ip-address-is-configured

OS: debian 10
SYSTEMD version: 241-7
Reproducibility: Always

Repro procedure (assume nfs server is up):
1. get debian 10 net install w/ only ssh-server as the only extra
2. install nfs-common
3. setup `/etc/fstab` include `x-systemd.requires=network-online.target, x-systemd.after=network-online.target`
--> login causes the mount to succeed, but actually did not succeed on boot, I use the following to check if mount worked on boot <--
4. setup mariadb to permit remote and require the mount (this is my way to checking externally )
5. reboot

Result:
1. journalctl --unit=my_nfs_mount
2. error on `network not reachable

Expectations:
1. can check from remote via mariadb login

What have you tried:
1. Everything that google gives:
a.) enable `systemd-networkd`
b.) enable `NetworkManager`
c.) include and play around with `fstab` mount options

Work around:
1. Check link: create another service that check the network and make the failing service 'require' the script.

**Comment by mbiebl** [comment]
Are you using systemd-networkd to configure your network? If so, can you share your configuration.
If not networkd, which network configuration system do you use?
Can you also post the output of `systemctl show network-online.target -p Wants -p After`
Which systemd version is this?

**Comment by yuwata** [comment]
Yeah, please fill all information in the issue format.

**Comment by poettering** [comment]
I think this answers most questions in this:

https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/

TLDR: what "online" means is to be defined by your network management solution. If it is the debian ifupdown then pleas ping the maintainers about that. If it's network manager then do "systemctl enable NetworkManager-wait-online", and if that is not sufficient, then pleas eping the maintainers of NM for helper. If it's systemd-networkd, then do "systemctl enable systemd-networkd".

Either way, this is not really a bug, but a support question I figure. Hence here's the canned response to that

<snipe>
*[Canned reply follows]*

This is a bug and feature request tracker, and not a support forum. In order to keep frustration at a minimum the submission form should have made this very clear.

If you have questions about systemd use or how to do or make use of certain things, please contact a more appropriate forum, for example the systemd mailing list:

https://lists.freedesktop.org/mailman/listinfo/systemd-devel

Thank you for understanding.
</snip>

Moreover, this is the upstream bug tracker, and you are using a heavily outdated systemd version hence the canned response regarding that:

<snip>
*[Canned reply follows]*

This is the upstream bug and feature request tracker of systemd. Please use this only for issues in the two most current upstream systemd versions. See this link for the list of current releases.

https://github.com/systemd/systemd/releases

For support for older versions please contact your distribution instead.

In order to keep frustration at a minimum the submission form tried to make this all very clear.
</snip>

**Comment by mbiebl** [comment]
If you are using Debian 10, I assume you use ifupdown as well.
In that case, you might consider using `auto` instead of `allow-hotplug` in your `/e/n/i` and/or enable ifupdown-wait-online.service

**Comment by multipool-readonly** [comment]
Why in the world is `setup /etc/fstab include x-systemd.requires=network-online.target, x-systemd.after=network-online.target` not the default?  How can NFS function if the network is not online?
